arch/cortex-m/src/nvic.rs

```diff
@@ -25,23 +25,23 @@ const NVIC_BASE_ADDRESS: StaticRef<NvicRegisters> =
 
 /// Clear all pending interrupts
 pub unsafe fn clear_all_pending() {
-    let nvic: &NvicRegisters = &*NVIC_BASE_ADDRESS;
+    let nvic: StaticRef<NvicRegisters> = NVIC_BASE_ADDRESS;
```

Comment:
I think the type annotation is technically superfluous.... but maybe useful for documentation?

Reply:
That's correct.